client/src/main/java/org/opensearch/geospatial/action/IpEnrichmentAction.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.core.action.ActionResponse;
+
+/**
+ * An ActionType registered on OpenSearch registry, for inter-cluster transportAction call,
+ * to resolve GeoLocation for IP String.
+ */
+public class IpEnrichmentAction extends ActionType<ActionResponse> {
```

Comment:
not sure why this action is in client? and its transport action is in root module. Please correct me if I am wrong here. 
One client implementation in Opensearch project I have seen is this: https://github.com/opensearch-project/ml-commons/tree/main/client .

Reply:
This is needed because during the plugin runtime, other plugin need to submit the `ActionType` as an identifier for one of the argument on `nodeClient.execute(ActionType, xxx, xx)`.
I have also checked on the above ML-common integration, indeed they also expose the respective action also `MLPredictionTaskAction`, but they decided on a separate module call `opensearch-ml-common` instead. 

I think both approach work, either we can keep everything in a jar and only expose `geoSpatial-client`, or we can follow the ML plugin approach, which expose TWO jars `ml-common` and `ml-client`, but essential contains the same classes.